fish.go

```diff
@@ -158,6 +160,25 @@ func (a *App) prepareFishFlags(
 	return completions
 }
 
+func fishAddFileFlag(
+	flag Flag,
+	completion *strings.Builder,
+) {
+	addFileExclusionFlag := true
+	if f, ok := flag.(GenericFlag); ok && f.TakesFile {
+		addFileExclusionFlag = false
+	}
+	if f, ok := flag.(StringFlag); ok && f.TakesFile {
+		addFileExclusionFlag = false
+	}
+	if f, ok := flag.(StringSliceFlag); ok && f.TakesFile {
+		addFileExclusionFlag = false
+	}
```

Comment:
```
switch v := flag.(type) {
   case StringFlag, GenericFlag, StringSliceFlag:
        if !v.TakesFile {
            completion.WriteString(" -f")
        }
}
```

Reply:
The comma separated type switch does not seem to work, fallthrough does also not work so I had to add dedicated statements for each type.